src/net_processing.cpp

```diff
@@ -1540,6 +1540,11 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
         }
     }
 
+    // Unknown types in the GetData stay in vRecvGetData and block any future
+    // getdata from this peer.
```

Comment:
Wouldn't this not only block future GetData, but also all other messages from this peer, because of the line `if (!pfrom->vRecvGetData.empty()) return true;` in ProcessMessages()?

Reply:
Indeed. In which case we might actually disconnect the peer, since there isn't anything that could happen on that connection anyway.

Reply:
This is also the only thing that prevents vRecvGetData from unbound growth.

Reply:
Perhaps I'd suggest adding this note to ``if (!pfrom->vRecvGetData.empty()) return true;``, otherwise it's not obvious why we should do this check before processing new messages.

Reply:
The comment there reads `// this maintains the order of responses`

Reply:
Which does not really explain that removing this line would enable DoS :)

Reply:
Thanks, done.